src/mesido/workflows/utils/adapt_profiles.py

```diff
@@ -76,97 +75,163 @@ def adapt_hourly_year_profile_to_day_averaged_with_hourly_peak_day(problem, prob
+            if day == max_day_hot // day_steps * day_steps or day == max_day_cold // day_steps * day_steps:
+                if day == max_day_hot // day_steps * day_steps:
+                    peak_day = max_day_hot
+                elif day == max_day_cold // day_steps * day_steps: # TODO: might have to fix this if the peak day is the same for heat and cold.
+                    peak_day = max_day_cold
+                if peak_day > day:
                     new_date_times.append(problem.io.datetimes[day * 24])
-                new_date_times.extend(problem.io.datetimes[max_day * 24 : max_day * 24 + 24])
-                if (day + day_steps - 1) > max_day:
-                    new_date_times.append(problem.io.datetimes[max_day * 24 + 24])
+                new_date_times.extend(problem.io.datetimes[peak_day * 24 : peak_day * 24 + 24])
+                if (day + day_steps - 1) > peak_day:
+                    new_date_times.append(problem.io.datetimes[peak_day * 24 + 24])
             else:
                 new_date_times.append(problem.io.datetimes[day * 24])
         new_date_times.append(problem.io.datetimes[-1] + datetime.timedelta(hours=1))
 
         new_date_times = np.asarray(new_date_times)
         parameters["times"] = [x.timestamp() for x in new_date_times]
-
-        for demand in demands:
-            var_name = f"{demand}.target_heat_demand"
+            
+        # heat_flag = False
+        # if heat_flag: # Case where there is only heat demands.
+        #     for day in range(0, nr_of_days, day_steps):
+        #         if day == max_day_hot // day_steps * day_steps: # Stops at the first day in the original discretization before the peak day
+        #             peak_day = max_day_hot
+        #             if peak_day > day: # Sitting behind the peak day
+        #                 new_date_times.append(problem.io.datetimes[day * 24]) # Just append a normal day. Not necessary if already at the peak day because it gets added in the next one if you are.
+        #             new_date_times.extend(problem.io.datetimes[peak_day * 24 : peak_day * 24 + 24]) # Appends the 24 hours of the peak day starting at the beginning of that peak day.
+        #             if (day + day_steps - 1) > peak_day: # Case where the end of the peak day does not coincide with the next step day. In that case, this "missing" day is added.
+        #                 new_date_times.append(problem.io.datetimes[peak_day * 24 + 24]) 
+        #         else:
+        #             new_date_times.append(problem.io.datetimes[day * 24]) # Normal day addition.
+        #     new_date_times.append(problem.io.datetimes[-1] + datetime.timedelta(hours=1)) # Adds one hour at the end of the year for some reason.
+            
+        #     new_date_times = np.asarray(new_date_times)
+        #     parameters["times"] = [x.timestamp() for x in new_date_times]
+
+        # Store the heat and cold target demands in the problem object, together with the now computed
```

Comment:
instead of rewriting, this part for when only heat demands exists, see the above comment.

Reply:
This is updated in the following comments.